src/java/org/apache/cassandra/locator/SimpleStrategy.java

```diff
@@ -100,12 +102,13 @@ public void validateOptions() throws ConfigurationException
     }
 
     @Override
-    public void maybeWarnOnOptions()
+    public void maybeWarnOnOptions(ClientState state)
     {
         if (!SchemaConstants.isSystemKeyspace(keyspaceName))
         {
             int nodeCount = StorageService.instance.getHostIdToEndpoint().size();
             // nodeCount==0 on many tests
+            Guardrails.minimumKeyspaceRF.guard(rf.fullReplicas, "minimum_keyspace_rf", false, state);
```

Comment:
The `what` argument refers to specific object that is evaluated by the guardrail, not the name of the guardrail. Here we are evaluating the replication factor of a specific keyspace, and `minimum_keyspace_rf` is the name of the guardrail. I would pass the name of the keyspace instead, and use a message provider such as, for example:
```java
isWarning ? format("The keyspace %s has a replication factor of %s, below the warning threshold of %s.",
                   what, value, threshold)
          : format("The keyspace %s has a replication factor of %s, below the failure threshold of %s.",
                   what, value, threshold)
```